x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/IdleConnectionEvictor.java

```diff
@@ -67,7 +67,7 @@ public synchronized void start() {
     private void startInternal() {
         logger.debug(() -> format("Idle connection evictor started with wait time: [%s] max idle: [%s]", sleepTime, maxIdleTime));
 
-        Scheduler.Cancellable task = threadPool.scheduleWithFixedDelay(() -> {
+        cancellableTask.set(threadPool.scheduleWithFixedDelay(() -> {
```

Comment:
I hit a unit test failure where `IdleConnectionEvictor::close` was called but the scheduled action before the task had been set.